vyper/parser/stmt.py

```diff
@@ -177,6 +177,9 @@ def assign(self):
             raise StructureException("Assignment statement must have one target", self.stmt)
         self.context.set_in_assignment(True)
         sub = Expr(self.stmt.value, self.context).lll_node
+        # Do not allow assignment to None
+        if sub.value is None:
+            raise InvalidLiteralException('None is not allowed as a constant, use a default value or built-in `reset()`.', self.stmt)
```

Comment:
Would it be preferable to throw a deprecation warning and an error, just an error, or just a warning?

Reply:
Just an error IMO

Reply:
Yeah that's my inclination as well.